gui/v5/backend.py

```
@@ -0,0 +1,645 @@
+            print(f"Controller path: {controller_path}")
+            print(f"Controller exists: {controller_path.exists()}")
+
+            if controller_path.exists():
+                try:
+                    # Change to the plotscode directory before running the script
+                    original_dir = os.getcwd()
+                    os.chdir(controller_path.parent)
+
+                    # Run the controller.py script to generate plots for both datasets
+                    print(f"Executing: {sys.executable} {controller_path}")
+                    result = subprocess.run(
```

Ruff (0.14.2)

`subprocess` call: check for execution of untrusted input

(S603)